project/SparkBuild.scala

```diff
@@ -443,6 +443,7 @@ object Hive {
         val relGFilePath = (gFilePath relativeTo sourceDir).get.getPath
         log.info("ANTLR: Grammar file '%s' detected.".format(relGFilePath))
         antlr.addGrammarFile(relGFilePath)
+        antlr.setLibDirectory(gFilePath.getParent)
```

Comment:
Brittle solution would be:

    val grammarDir = sourceDir / "org" / "apache" / "spark" / "sql" / "catalyst" / "parser"
    antlr.setLibDirectory(grammarDir.getPath)
    grammarFileNames.foreach { gFileName =>
        // We construct the entire path in order to make sure the file exists.
        val relGFilePath = ((grammarDir / gFileName) relativeTo sourceDir).get.getPath
        log.info("ANTLR: Grammar file '%s' detected.".format(relGFilePath))
        antlr.addGrammarFile(relGFilePath)
      }